src/main/java/org/reaktivity/nukleus/maven/plugin/internal/ast/AstType.java

```diff
@@ -123,6 +128,16 @@ public boolean isUnsignedInt()
         return this == UINT8 || this == UINT16 || this == UINT24 || this == UINT32 || this == UINT64 || this == VARBYTEUINT32;
     }
 
+    public boolean isDynamicType()
+    {
+        return !isSignedInt() && !isUnsignedInt() && this != OCTETS && this != STRING && this != STRING8 && this != STRING16 &&
+            this != STRING32 && this != ARRAY && this != FLYWEIGHT && this != LIST && this != LIST0 && this != LIST8 &&
+            this != LIST32 && this != VARIANT && this != VARIANT_OF && this != VARIANT_ARRAY && this != VARIANT_ARRAY8 &&
+            this != VARIANT_ARRAY16 && this != VARIANT_ARRAY32 && this != MAP && this != MAP8 && this != MAP16 &&
+            this != MAP32 && this != BOUNDED_OCTETS  && this != BOUNDED_OCTETS8 && this != BOUNDED_OCTETS16 &&
+            this != BOUNDED_OCTETS32;
```

Comment:
Given this is a long list of types, we would probably be better of adding these types into an identity hash set, then changing all the `this != ...` checks to instead ensure `this` is not present in that identity hash set of static types.